numba/np/arraymath.py

```diff
@@ -3899,6 +3899,19 @@ def impl(a, dtype=None):
     return impl
 
 
+@overload(np.asfarray)
+def np_asfarray(a, dtype=types.float64):
```

Comment:
think this ought to be `dtype=np.float64`?

Reply:
and then something check that the thing passed in is actually a dtype or can be accepted and converted to one?

Reply:
Is the check necessary? if `dtype` is not a `subdtype` of `np.inexact`, then, `dtype` will be assigned to `types.float64`.

Reply:
ok, I think the check can be skipped as the `as_dtype` will pick up nonsense types.